### PR TITLE
feat: Implement MariaDB `ALTER TABLE` syntax

### PR DESCRIPTION
**Pull request by LeoniePhiline**
# Implement MariaDB & MySQL `ALTER TABLE` syntax

Replacing the accidentally merged #98.

This PR is a draft just to show I'm working on that, since now other PRs have been merged and no parser code clashes await.

## Fixes:
- https://github.com/DerekStride/tree-sitter-sql/issues/82
- https://github.com/DerekStride/tree-sitter-sql/issues/83
- https://github.com/DerekStride/tree-sitter-sql/issues/89
- #71 

## PR contents:

- [fix: Lift `ALTER TABLE` combination restriction (only Postgres)](https://github.com/DerekStride/tree-sitter-sql/commit/be20ed053a6d0a6125e485f4796a488f91df987d)
  Resolves comment: https://github.com/DerekStride/tree-sitter-sql/issues/82#issuecomment-1416227381
- [feat: Add MariaDB/MySQL `ALTER TABLE ... CHANGE ...`](https://github.com/DerekStride/tree-sitter-sql/pull/99/commits/1124c5151320e6c1107ef7afc7f61b91690f0ebe)
  Fixes: https://github.com/DerekStride/tree-sitter-sql/issues/82
- [feat: Add MariaDB/MySQL `ALTER TABLE ... MODIFY ...`](https://github.com/DerekStride/tree-sitter-sql/pull/99/commits/409be47ae61daeac205be5448b1b69fb702e4915)
  Fixes: https://github.com/DerekStride/tree-sitter-sql/issues/89
- [feat: Add MariaDB/MySQL `ADD|CHANGE|MODIFY ... FIRST|AFTER ...`](https://github.com/DerekStride/tree-sitter-sql/pull/99/commits/393e0d35aad922a3bf4faa4ceccad2e401f7b2ab)
  Fixes: https://github.com/DerekStride/tree-sitter-sql/issues/83

### CODE REVIEW

**Comment by LeoniePhiline** [comment]
PTAL

File: grammar.js [hunk 1]
```diff
@@ -698,37 +701,37 @@ module.exports = grammar({
...
-              choice(
-                $.add_column,
-                $.alter_column,
-                $.drop_column,
-              )
+              $._alter_specifications
             )
           )
         ),
-        // some operations may not be chained to others
```
Comment (DerekStride) [inline grammar.js:717]: Is this still valid or was that just an artifact of another dialect of SQL?

**Review by DerekStride — APPROVED** [review]
One question otherwise LGTM

File: grammar.js (hunk 1 above)
Comment (LeoniePhiline) [reply grammar.js:717]: [fix: Lift ALTER TABLE combination restriction (only Postgres)](https://github.com/DerekStride/tree-sitter-sql/commit/be20ed053a6d0a6125e485f4796a488f91df987d)
Resolves comment: https://github.com/DerekStride/tree-sitter-sql/issues/82#issuecomment-1416227381

**Pull request merged**